doc/howto/write_an_alphabet/index.md

```diff
@@ -0,0 +1,149 @@
+
+It describes the requirements for types that are comparable via `<`, `<=`, `>` and `>=`.
+This is useful so that you can sort a text over the alphabet, for example.
+Additionally, std::StrictTotallyOrdered is again a refinement of std::EqualityComparable,
+which requires the `==` and `!=` operators, so let's first implement those with free member functions.
+\snippet dna2_semialphabet.cpp equality
+
+\assignment{Excercise}
+Implement the inequality operator (!=) for `dna2`.
+\endassignment
+\solution
+\snippet dna2_semialphabet.cpp inequality
+\endsolution
+
+We see that our type now models std::EqualityComparable, which is a prerequisite of std::StrictTotallyOrdered.
+\snippet dna2_comparison_operators.cpp equality_comparable_concept
+
+\assignment{Excercise}
+Implement the four comparison operators *as free member functions* and verify that your type models
+std::StrictTotallyOrdered.
+\endassignment
+\solution
+\snippet dna2_comparison_operators.cpp comparison
+\endsolution
+
+## Semialphabet
+
+Let's move on to the more interesting concepts. seqan3::Semialphabet requires the *rank interface*
+that we have introduced in the [alphabet tutorial](\ref tutorial_alphabets). If we define the member functions
+`to_rank`, `assign_rank` and `alphabet_size` for our type, SeqAn automatically exposes them to the respective global
+functions `seqan3::to_rank`, `seqan3::assign_rank_to` and `seqan3::alphabet_size`.
+Note that `assign_rank` is only required for seqan3::WritableSemialphabet.
+\snippet dna2_semialphabet.cpp semialphabet
+
+As you can see from the `static_assert` our dna2 alphabet now models seqan3::Semialphabet and
+seqan3::WritableSemialphabet:
+\snippet dna2_semialphabet.cpp writable_semialphabet_concept
+
+With this we can automatically use some free functions on our dna2 alphabet. If you are interested in how this works,
+read our section on [Customisation points](\ref about_customisation). Here is an example:
+\snippet dna2_semialphabet.cpp free_functions
+
+## Alphabet
+
+Now that you have a feeling for concepts, have a look at seqan3::Alphabet and seqan3::WritableAlphabet and make
+your type also model these concepts. If you keep the suggested namings, SeqAn will automatically
+expose the members to the respective global functions that model the concept.
+
+\assignment{Excercise}
+1. Implement the member function `char to_char(void)` which returns the char 'S' or 'W' depending on the rank value.
+2. Implement the member function `dna2 & assign_char(char const)`.
+3. seqan3::WritableAlphabet additionally requires that the type has a member function `bool char_is_valid`
+   which can be used to determine whether a given character is an actual representation of the alphabet.
+   For the `dna2` alphabet 'S' and 'W' as well as 's' and 'w' should be valid, but nothing else.
+   <br>
+   Implement the static member function `bool char_is_valid(char const)`.
```

Comment:
No, this is not required, it is optional. I would maybe skip it in the first example. Or explicitly state that it is optional.